lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+/// <reference types="@panva/jose" />
+/// <reference types="node" />
+
+/**
+ * @see https://github.com/panva/node-openid-client/blob/master/docs/README.md
+ */
+declare module 'openid-client' {
+  import { JWKS } from '@panva/jose'
+  import { IncomingMessage } from 'http'
+  import { Http2ServerRequest } from 'http2'
+
+  /**
+   * @see https://medium.com/@darutk/diagrams-of-all-the-openid-connect-flows-6968e3990660
+   */
+  enum AuthResponseType {
```

Comment:
If we're defining ambient enums in the declaration file, they'll also need to exist in the javascript to be useable

Reply:
May be more suitable to use a string literal if we want to restrict the set of auth types without needing to access an existing enum in JS

Reply:
Agreed, ergo please change all enums to `type Foo = 'a' | 'b' | 'c'` instead